spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -580,7 +598,7 @@ public Void doInRabbit(Channel channel) throws Exception {
 						return null;
 					}
 				});
-				this.usingFastReplyTo = true;
+				return true;
 			}
 			catch (Exception e) {
 				if (this.replyAddress != null) {
```

Comment:
this line will be invoked only if replyAddress is null (condition at line 591) so replyAddress will be always null (assuming that this is thread safe)